ovn_k8s/processor/policy_processor.py

```diff
@@ -28,34 +30,192 @@ def __init__(self, pool):
+        pod_ns = pod_data['metadata']['namespace']
+
+        log.dbg("Pod: %s (Namespace: %s): applying ACLs..." % (
+            pod_name, pod_ns))
+        # Check for a namespace event. This might mean that there has
+        # been a translation from isolated to non-isolated and therefore
+        # all we have to do would be whitelisting traffic
+        self._apply_pod_ns_acls(pod_data, pod_events)
+        if kubernetes.is_namespace_isolated(
+                variables.K8S_API_SERVER, pod_ns):
+            # If we are here the namespace is isolated, and policies must
+            # be translated into acls
+            self.mode.apply_pod_policy_acls(pod_data, policies)
+        log.dbg("Pod: %s (Namespace: %s): ACLs applied" % (pod_name, pod_ns))
 
     def process_events(self, events):
         log.dbg("Processing %d events from queue" % len(events))
-        for event in events[:]:
+        affected_pods = {}
+        pod_events = {}
+        for event in events:
             if isinstance(event, processor.NSEvent):
                 # namespace add -> create policy watcher
                 # namespace delete -> destory policy watcher
                 # namespace update -> check isolation property
-                self._process_ns_event(event)
-            events.remove(event)
+                self._process_ns_event(event, affected_pods, pod_events)
+            if isinstance(event, processor.NPEvent):
+                # policy add -> create ACLs for affected pods
+                # policy delete -> remove ACLs for affected pods
+                self._process_np_event(event, affected_pods, pod_events)
+            if isinstance(event, processor.PodEvent):
+                # relevant policies must be applied to pod
+                # check policies that select pod in from clause
+                self._process_pod_event(event, affected_pods, pod_events)
 
-        for event in events:
-            log.warn("Event %s from %s was not processed. ACLs might not be "
-                     "in sync with network policies",
-                     event.event_type, event.source)
-        else:
-            log.info("Event processing terminated.")
+        ns_policy_map = {}
+        for pod_id, pod_data in affected_pods.items():
+            pod_ns = pod_data['metadata']['namespace']
+            policies = ns_policy_map.get(
+                pod_ns, kubernetes.get_network_policies(
+                    variables.K8S_API_SERVER,
+                    pod_data['metadata']['namespace']))
+            ns_policy_map[pod_ns] = policies
+            log.dbg("Rebuilding ACL for pod:%s because of:%s" %
+                    (pod_id, "; ".join(['%s from %s' % (event.event_type,
+                                                        event.source)
+                                        for event in pod_events[pod_id]])))
+            self._apply_pod_acls(pod_data, pod_events[pod_id], policies)
+
+        log.info("Event processing terminated.")
```

Comment:
oh yes, and it's not the only one.
I will remove this as a starter, but I was thinking on a different PR to remove log clutter.